Few fixes

ImMorpheus opened the pull request:
Few fixes (wip)

jamierocks commented on src/html/downloads.html:
```diff
@@ -39,7 +39,7 @@
         <div class="container">
             <div class="logo">
-                <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt="Spongie reverse"/>
```
This really isn't required, not going to help accessibility at all.

jamierocks commented on src/html/include/base.html:
```diff
@@ -55,7 +55,7 @@
         <div id="sp-logo-container" class="page-scroll">
             <a class="logo" href="#top">
-                <img src="assets/img/icons/spongie-mark.svg" style="height: 40px">
+                <img src="assets/img/icons/spongie-mark.svg" alt="Spongie" style="height: 40px">
```
This really isn't required, not going to help accessibility at all.

jamierocks commented on src/js/downloads/Downloads.vue:
```diff
@@ -30,7 +30,7 @@
                 <div class="col-lg-5 col-md-6">
                     <div class="logo">
-                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt="Spongie reverse" />
```
This really isn't required, not going to help accessibility at all.

jamierocks commented on src/js/downloads/Main.vue:
```diff
@@ -30,7 +30,7 @@
                 <div class="col-lg-5 col-md-6">
                     <div class="logo">
-                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt="Spongie reverse"/>
```
This really isn't required, not going to help accessibility at all.

ImMorpheus commented:
The alt attribute is required for img tag. However if you really don't like the content we can use empty alt.

Edit: I forgot about the ocd list, thanks for that

jamierocks commented:
Yes they should have an empty alt

jamierocks commented:
As of HTML5 self-closed tags (such as img) no longer need to use specific syntax (`/>`), and can just be left as a single tag.

ImMorpheus commented:
Note: currently I had to add in the CSP connect-src for the download page and frame-src for kiwilrc. Everything works, if we want to make it more restrictive we can move the fonts on the server and use font-src 'self'

jamierocks commented:
> if we want to make it more restrictive we can move the fonts on the server and use font-src 'self'

That'd be overkill

jamierocks commented:
Not to demean your pr, but these changes are the type of things that an OCD issue exists for in every SpongePowered repository ([SpongeHome issue](https://github.com/SpongePowered/SpongeHome/issues/104)) for. 

Perhaps exception can be made of this pr, in light of the number of issues resolved - however in future that is certainly where these issues should be documented.

ImMorpheus commented:
Yeah, you've already said that (and I added the issues to the OCD list), but if you really want to fix them I can delete the pull request and no 'exception' will be made. 
That being said, I fixed a few things and created this pull because I knew changes would be over 20 lines. [Thank goodness small pull request get closed and people informed they should use the OCD.](https://github.com/SpongePowered/SpongeHome/pull/122)

jamierocks commented:
> Thank goodness small pull request get closed and people informed they should use the OCD.

In regards to that pr, I am not sure why that was merged and not directed to the OCD issue. It should have been.

jamierocks commented on controllers/header.go:
```diff
@@ -26,20 +26,17 @@
-	"style-src 'self' 'unsafe-inline' https:; "+
-	"script-src 'self' 'unsafe-eval' https://cdnjs.cloudflare.com https://www.google-analytics.com; "+
-	"frame-ancestors 'none'")*/
+	header.Set("Content-Security-Policy", "default-src 'none'; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://cdnjs.cloudflare.com/ ; style-src 'self' 'unsafe-inline' https://cdnjs.cloudflare.com; img-src 'self'; font-src https://cdnjs.cloudflare.com/ https://fonts.googleapis.com/; connect-src 'self' https://dl-api.spongepowered.org/; frame-src https://kiwiirc.com; frame-ancestors 'none'; u ...
```
Can we get this split over a few lines, to promote readability?

jamierocks commented on src/html/downloads.html:
```diff
@@ -39,7 +39,7 @@
         <div class="container">
             <div class="logo">
-                <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt=""/>
```
Change `/>` for `>` as per HTML5.

jamierocks commented on src/html/include/base.html:
```diff
@@ -129,7 +129,7 @@
                 <p id="fastly">Accelerated by<br>
                     <a href="https://www.fastly.com">
-                        <img src="assets/img/fastly.png" alt="Fastly" />
+                        <img src="assets/img/fastly.png" alt="Fastly" height="45" width="100" />
```
Change `/>` for `>` as per HTML5.

jamierocks commented on src/js/downloads/Downloads.vue:
```diff
@@ -30,7 +30,7 @@
                 <div class="col-lg-5 col-md-6">
                     <div class="logo">
-                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt="" />
```
Change `/>` for `>` as per HTML5.

jamierocks commented on src/js/downloads/Main.vue:
```diff
@@ -30,7 +30,7 @@
                 <div class="col-lg-5 col-md-6">
                     <div class="logo">
-                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" />
+                        <img src="assets/img/icons/spongie-mark-reverse-dark.svg" alt=""/>
```
Change `/>` for `>` as per HTML5.

jamierocks commented on README.md:
```diff
@@ -1,20 +1,19 @@
 ### Branches
 
-- `master` is hooked up to [www staging](https://www-staging.spongepowered.org/)
 - `production` is hooked up to [www prod](https://www.spongepowered.org/)
```
The production branch is no longer used - master is now `www prod`.

jamierocks commented on .gitignore:
```diff
@@ -80,13 +80,13 @@ lib-cov
 coverage
 
-# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
+# Grunt intermediate storage (https://gruntjs.com/creating-plugins#storing-task-files)
```
May as well remove Grunt from the gitignore - we don't use it.

ImMorpheus replied:
Is that okay ? You created that gitignore with gitignore.io, it is basically a preset which handle multiple developers using different ide/tools/etc

jamierocks replied:
Yeah - Grunt is a buildtool, we use Gulp so Grunt will never become part of our dev env.

ImMorpheus replied:
Done

ImMorpheus commented on .gitignore:
```diff
@@ -80,13 +80,13 @@ lib-cov
 coverage
 
-# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
+# Grunt intermediate storage (https://gruntjs.com/creating-plugins#storing-task-files)
```
Shall I open an issue if I was able to return the nginx version in the home or shall I contact you privately ? (Yeah I know it's off topic, I didn't know where to ask this)

stephan-gh commented on .gitignore:
```diff
@@ -80,13 +80,10 @@ lib-cov
 # Coverage directory used by tools like istanbul
 coverage
 
-# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
```
There is no reason to remove this. The gitignore is based on a template and since this isn't breaking anything in our project I see no reason to remove it.

ImMorpheus replied:
I knew it would end up this way

stephan-gh commented on .gitignore:
```diff
@@ -80,13 +80,10 @@ lib-cov
 .lock-wscript
 
-# Compiled binary addons (http://nodejs.org/api/addons.html)
+# Compiled binary addons (https://nodejs.org/api/addons.html)
```
Feel free to suggest this change in the upstream repository: https://github.com/github/gitignore/blob/master/Node.gitignore#L32. I don't see a reason to change it here.

stephan-gh commented on controllers/header.go:
```diff
@@ -26,20 +26,23 @@
-	"style-src 'self' 'unsafe-inline' https:; "+
-	"script-src 'self' 'unsafe-eval' https://cdnjs.cloudflare.com https://www.google-analytics.com; "+
-	"frame-ancestors 'none'")*/
+	header.Set("Content-Security-Policy", "default-src 'none'; " +
```
Please move this to a separate PR, it needs careful testing and can't be accepted together with the other minor changes.

ImMorpheus replied:
Oh ok, I didn't open a new pull because it was a rather small fix (in terms of lines)

stephan-gh commented on controllers/header.go:
```diff
@@ -49,3 +49,54 @@ func AddHeaders(resp http.ResponseWriter) {
 		header.Add(fastly.SurrogateControlHeader, cache.SurrogateStaticContentOptions)
 	}
 }
+/*
```
I think this is not what you wanted to do :)

ImMorpheus replied:
Yeah, I noticed that soon after

kashike commented:
You should not be changing any of the sponsor images. :no_good_man:

ImMorpheus commented:
I just compressed the images, can't I do that ? (Not a sensitive change, but now they are compressed)
Also, is the backgroud included in the group of images I shouldn't change ?

stephan-gh commented:
@ImMorpheus It's probably better if we keep them as-is, because they were not made by us. Regarding the background, it seems like your optimized image has the same size as the original one?

ImMorpheus commented:
Ok, I didn't know that. Consider it fixed
The backgroud image is 1% smaller, we are talking about nearly 500B (that's why I thought it was not a big deal)

stephan-gh commented:
@ImMorpheus 500 bytes isn't really worth changing the image, that is almost nothing.

stephan-gh commented on src/html/include/base.html:
```diff
@@ -129,7 +129,7 @@
                 <p id="fastly">Accelerated by<br>
                     <a href="https://www.fastly.com">
-                        <img src="assets/img/fastly.png" alt="Fastly" />
+                        <img src="assets/img/fastly.png" alt="Fastly" height="45" width="100">
```
Why did you add an explicit width/height?

ImMorpheus replied:
If height and width are set, the space required for the image is reserved when the page is loaded. Thus, the page loads faster.

ImMorpheus commented:
Yeah, I compressed the images all together and there was a compression of more than 20% with some of the others. It just happened to be there and was compressed

stephan-gh commented:
@ImMorpheus That's fine but please revert the change. We shouldn't require all clients to download the whole image again just for 500 bytes.

ImMorpheus commented:
Done
